irc/accounts.go

```diff
@@ -1047,6 +1047,17 @@ func (am *AccountManager) AuthenticateByPassphrase(client *Client, accountName s
 	return err
 }
 
+func (am *AccountManager) AllNicks() []string {
+	am.RLock()
+	defer am.RUnlock()
+
+	nicks := make([]string, 0, len(am.nickToAccount))
```

Comment:
These are the casefolded nicks, and it's all nicks as opposed to all accounts.

I think since grouped nicks are basically deprecated (the recommended configuration now is nick-equals-account), it would be better to return a list of all uncasefolded account names. This would also have closer parity with `AllChannels()` (you can basically copy that implementation).

Thoughts?

Reply:
Hmm. I figured that grouped nicknames should be returned as part of this, since they are registered nicknames, no?

To resolve the casefolding issue, I could read the names from the account itself rather than the map keys.

Reply:
Yeah, that's fair, let's return all the registered nicknames in their uncasefolded forms.

Reply:
I changed this to read from the database instead.